api4/group_test.go

```diff
@@ -328,6 +328,19 @@ func TestLinkGroupTeam(t *testing.T) {
 	groupTeam, response, _ := th.Client.LinkGroupSyncable(g.Id, th.BasicTeam.Id, model.GroupSyncableTypeTeam, patch)
 	assert.Equal(t, http.StatusCreated, response.StatusCode)
 	assert.NotNil(t, groupTeam)
+
+	gid := model.NewId()
+	g2, app2Err := th.App.CreateGroup(&model.Group{
+		DisplayName: "dn_" + gid,
+		Name:        model.NewString("name" + gid),
+		Source:      model.GroupSourceCustom,
+		Description: "description_" + gid,
+		RemoteId:    model.NewString(model.NewId()),
+	})
+	assert.Nil(t, app2Err)
+
+	_, _, err = th.Client.LinkGroupSyncable(g2.Id, th.BasicTeam.Id, model.GroupSyncableTypeTeam, patch)
+	require.Error(t, err)
```

Comment:
For this test, I think it would be good to also verify that the correct error is returned, otherwise this test could give a false positive if the implementation changes in the future.

Reply:
I also think based on the change I've recommended in the conditional we should test:
1. source is custom
2. source is something random

Reply:
Unfortunately we can't test with a random source because whenever we create a group it ensures the source is either `ldap` or `custom`

Reply:
> For this test, I think it would be good to also verify that the correct error is returned, otherwise this test could give a false positive if the implementation changes in the future.

Done